src/components/context/cartAndWhishlistContext.jsx

```diff
@@ -0,0 +1,143 @@
+import { createContext, useReducer, useContext } from "react";
+import {
+  cartWishListReducer,
+  alreadyInWishlist,
+  alreadyInCart,
+} from "../index";
+import axios from "axios";
+import { useNavigate } from "react-router-dom";
+
+const CartAndWishlistContext = createContext(null);
+let encodedToken =
```

Comment:
Why is the token written here? Because I guess we have to fetch it after successful login.